### PR TITLE
Enhance population tab

### PR DESCRIPTION
**Pull request by rafa-garcia**
CONTEXT: This changeset adds a number of enhancements to the population table, as follows:

  * Religion
      - [x] Display year of data, if available, as secondary data
      - [x] If data is available, display percentage as with no decimal places
      - [x] Display top 5 entries, ranked in order

  * Language
      - [x] If available, show primary "note" field
      - [x] Display year of data, if available, as secondary data
      - [x] If data is available, display percentage as with no decimal places
      - [x] Display top 5 entries, ranked in order

  * Rule of Law
      - [ ] Display year of data, as secondary data 
      - [x] Show rank, e.g. 5 of 113, as opposed to score
      - [x] Update heading to "Rule of Law ranking"

### Workflow

- [X] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-1346
- [X] Jira ticket has the correct status.
- [X] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [X] Explains how to test locally, including how to set up appropriate data
- [X] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [X] Frontend assets have been re-compiled.

### Merging

- [X] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Comment by rafa-garcia** [comment]
image

File: react-components/src/components/CompareMarkets/TabConfiguration.jsx [hunk 1]
```diff
@@ -28,23 +28,85 @@ const rankOutOf = (data, key) => {
...
+        {year && !note &&(
+          <div className="body-m text-black-60 display-year">{year}</div>
+        )}
+        {!year && note && (
+          <div className="body-m text-black-60 display-note">{capitalize(note)}</div>
+        )}
+        {year && note && (
+          <div className="body-m text-black-60 display-note">{year}. {capitalize(note)}</div>
+        )}
+      </>
```
Comment (bobmeredith) [inline react-components/src/components/CompareMarkets/TabConfiguration.jsx:72]: This is a bit repetitive.  How about something like :
`<div className="body-m text-black-60 display-note">{year}{year && note && ". "}{capitalize(note)}</div>`

**Review by bobmeredith — COMMENTED** [review]
Nice - but I have a suggestion

**Review by bobmeredith — APPROVED** [review]
LGTM

**Pull request merged**